jwst/extract_1d/extract.py

```diff
@@ -325,6 +338,12 @@ def get_extract_parameters(
                     else:
                         extract_params['src_coeff'] = aper.get('src_coeff')
                         extract_params['bkg_coeff'] = aper.get('bkg_coeff')
+                        if extract_params['bkg_coeff'] is not None:
+                            extract_params['subtract_background'] = True
+                            extract_params['bkg_fit'] = aper.get('bkg_fit', 'poly')
+                        else:
+                            extract_params['bkg_fit'] = None
+                            extract_params['subtract_background'] = False
```

Comment:
Same here.  Not unit tested, but perhaps a small test could target some of this.

Reply:
Again, this is at least tested in the regression tests, where the NIRISS SOSS extract1d ref file specifies `bkg_coeff` values, but none of the other spectral modes do, so both branches do get tested in regtest. But unit test coverage certainly couldn't hurt.